rotkehlchen/exchanges/binance.py

```diff
@@ -589,6 +607,184 @@ def _query_lending_balances(
 
         return balances
 
+    def _api_query_list_paginated_within_time_delta(
+            self,
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+            time_delta: int,
+            api_type: BINANCE_API_TYPE,
+            method: Literal['lending/union/interestHistory'],
+            additional_options: Optional[dict] = None,
+    ) -> list[dict[str, Any]]:
+        """Request via `_api_query_list_within_time_delta` using `current` param for pagination"""
+        data = []
+        current_page = 1
+        while True:
+            if additional_options:
+                additional_options.update({'current': current_page})
+
+            response = self._api_query_list_within_time_delta(
+                api_type=api_type,
+                start_ts=start_ts,
+                end_ts=end_ts,
+                time_delta=time_delta,
+                method=method,
+                additional_options=additional_options,
+            )
+            # we get an empty response, and `current` is at least 1
+            if len(response) == 0 and current_page >= 1:
+                break
+
+            data.extend(response)
+            current_page += 1
+
+        return data
+
+    def _query_lending_interests_history(
+            self,
+            cursor: 'DBCursor',
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+    ) -> None:
+        """Queries Binance lending interest history, transforms it
+        into `HistoryBaseEntry` objects and save it in the database.
+
+        "DAILY" -> flexible savings.
+        "CUSTOMIZED_FIXED" -> fixed savings.
+        "ACTIVITY" -> fixed savings but as a result of token partners.
+
+        Lending Interest History Documentation:
+        https://binance-docs.github.io/apidocs/spot/en/#get-interest-history-user_data-2
+
+        May raise:
+        - RemoteError
+        """
+        ranges = DBQueryRanges(self.db)
+        range_query_name = f'{self.location}_history_events_{self.name}'
```

Comment:
I would use a different name here since history_events is a wide tag.

```suggestion
        range_query_name = f'{self.location}_lending_history_{self.name}'
```

Also you need to make sure that when deleting the information from binance and binance us this query range also gets deleted. It happens in the dbhandler file. I believe it does because it searches by self.location but can you double check it?